dra: update for Kubernetes 1.28 (#41856)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md

<!-- overview -->

> {{< feature-state for_k8s_version="v1.28" state="alpha" >}}

Reviewer: ```suggestion
{{< feature-state for_k8s_version="v1.27" state="alpha" >}}
```
This feature reached alpha in v1.27

running Pods. For more information on the gRPC endpoints, see the
[resource allocation reporting](/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#monitoring-device-plugin-resources).

## Pre-scheduled Pods

> When creating a Pod with `nodeName` already set, the scheduler gets bypassed.

Reviewer: ```suggestion
When you - or another API client - create a Pod with `.spec.nodeName` already set, the
scheduler gets bypassed.
```

If some ResourceClaim needed by that Pod does not exist yet, is not allocated

> or not reserved for the Pod, then the kubelet will fail to run the Pod and

Reviewer 2: it will be great to add instruction on how to unstuck it. I think simply deleting the pod will do. The only thing - mention that if Pod is part of the replicaset, another instance may be created with the same issue.

Author: There are numerous reasons for what might be wrong, so listing all possible remediation here will not be possible. The goal in this section is more about raising awareness of the problem ("don't do this"!) and explain that some automatic mitigation is available now through the kube-controller-manager changes.

re-check periodically because those requirements might still get fulfilled
later.

Such a situation can also arise when support for dynamic resource allocation
was not enabled in the scheduler at the time when the Pod got scheduled
(version skew, configuration, feature gate, etc.). kube-controller-manager
detects this and tries to make the Pod runnable by triggering allocation and/or
reserving the required ResourceClaims.

> However, it is better to avoid this because a Pod which is assigned to a node

Reviewer: (nit)
```suggestion
However, it is better to avoid this because a Pod that is assigned to a node
```

blocks normal resources (RAM, CPU) that then cannot be used for other Pods
while the Pod is stuck. To make a Pod run on a specific node while still going

> through the normal scheduling flow, create the Pod with a node selector that

Reviewer 3: ```suggestion
through the normal scheduling flow, create the Pod with a `nodeSelector` that
```

Not quite sure if this is correct but you capitalize other api fields or used code style for those.  Thought it looked a bit off.

Author: "a node selector" refers to the general concept here, not the specific field. That is then shown in the example. I think using plain English is fine here and also used elsewhere.

> matches exactly the desired node:

Reviewer: ```suggestion
exactly matches the desired node:
```

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod-with-cats
spec:
  nodeSelector:
    kubernetes.io/hostname: name-of-the-intended-node
  ...
```

> 

Reviewer: Optionally:
```suggestion

You may also be able to mutate the incoming Pod, at admission time, to unset the `.spec.nodeName`
field and to use a node selector instead.

```
